core/extraction/function_extractor.py

```
@@ -217,6 +166,65 @@
+    ) -> List[ExtractedFunction]:
+        """Extract function definitions from AST nodes."""
+        functions: List[ExtractedFunction] = []
+
+        # Ensure we process all nodes
+        nodes_to_process = [nodes] if isinstance(nodes, ast.AST) else nodes
+        self.logger.info("Starting function extraction.")
+        for node in ast.walk(
+            nodes_to_process[0] if nodes_to_process else ast.Module(body=[])
+        ):
+            if isinstance(node, (ast.FunctionDef, ast.AsyncFunctionDef)):
+                if self._should_process_function(node):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)